dashbuilder/dashbuilder-client/dashbuilder-displayer-client/src/main/java/org/dashbuilder/displayer/client/widgets/DisplayerTypeSelector.java

```diff
@@ -42,26 +43,29 @@
     DisplayerType selectedType = DisplayerType.BARCHART;
     DisplayerSubtypeSelector subtypeSelector;
     Event<DisplayerTypeSelectedEvent> typeSelectedEvent;
+    RendererManager rendererManager;
 
     @Inject
     public DisplayerTypeSelector(View view,
                                  DisplayerSubtypeSelector subtypeSelector,
-                                 Event<DisplayerTypeSelectedEvent> typeSelectedEvent) {
+                                 Event<DisplayerTypeSelectedEvent> typeSelectedEvent,
+                                 RendererManager rendererManager) {
         this.view = view;
         this.subtypeSelector = subtypeSelector;
         this.typeSelectedEvent = typeSelectedEvent;
+        this.rendererManager = rendererManager;
         view.init(this);
         view.clear();
-        view.show(DisplayerType.BARCHART);
-        view.show(DisplayerType.PIECHART);
-        view.show(DisplayerType.LINECHART);
-        view.show(DisplayerType.AREACHART);
-        view.show(DisplayerType.BUBBLECHART);
-        view.show(DisplayerType.METERCHART);
-        view.show(DisplayerType.METRIC);
-        view.show(DisplayerType.MAP);
-        view.show(DisplayerType.TABLE);
-        view.show(DisplayerType.SELECTOR);
+        verifySupportAndShow(DisplayerType.BARCHART);
```

Comment:
Optinal comment: I'd recommend to using something like
```java
Arrays.stream(DisplayerType.values())
    .filter(type -> rendererManager.isTypeSupported(type))
    .forEach(type -> view.show(type));
```
Apart from being a bit shorter, you don't have to enumerate all the values, making the code more resistant to changes in the enum.

Reply:
@jhrcek Thanks for the suggestion! Refactored to:

~~~
Arrays.stream(DisplayerType.values())
      .filter(rendererManager::isTypeSupported)
      .forEach(view::show);
~~~